backend/workstation/ws_controllers.js

```diff
@@ -0,0 +1,224 @@
+const axios = require("axios");
+const prisma = require("../prisma/prisma_client");
+require("dotenv").config();
+
+//Create New File
+exports.createNewFile = async (req, res) => {
+  const { groupId } = req.params;
+  const { newFileName } = req.body;
+  const userId = req.user.id;
+
+  if (!newFileName) {
+    return res.status(400).json({
+      error: "File name is required",
+    });
+  }
+
+  try {
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        members: true,
+      },
+    });
+    if (!group) {
+      return res.status(404).json({
+        error: "Group not found",
+      });
+    }
+    const isMember = group.members.some((member) => member.userId === userId);
+    if (!isMember && group.creatorId !== userId) {
+      return res.status(403).json({
+        error: "You do not have access to this group",
+      });
+    }
+    const newFile = await prisma.file.create({
+      data: {
+        fileName: newFileName,
+        user: {
+          connect: {
+            id: userId,
+          },
+        },
+        codeGroup: {
+          connect: {
+            id: parseInt(groupId),
+          },
+        },
+      },
+    });
+    return res.status(201).json({
+      message: "File created successfully",
+      file: newFile,
+    });
+  } catch (error) {
+    res.status(500).json({
+      error: "Error creating file.",
+    });
+  }
+};
+
+//Get File Details
+exports.getFileDetails = async (req, res) => {
+  const { groupId, fileId } = req.params;
+  const userId = req.user.id;
+
+  try {
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        members: true,
+      },
+    });
+
+    if (!group) {
+      return res.status(404).json({
+        error: "Group not found",
+      });
+    }
+
+    const isMember = group.members.some((member) => member.userId === userId);
+    if (!isMember && group.creatorId !== userId) {
+      return res.status(403).json({
+        error: "you do not have access to this group",
+      });
+    }
```

Comment:
Since this is common code that is repeated in several functions, I would extract this to a helper function that you can reuse.

Reply:
Thanks for the feedback. I have updated this here: https://github.com/izuchukwumba/peered/pull/7/commits/1d04a3e6e4fc6218291de43471fdebfdc707b1d1